src/TuDa.CIMS.Api/Controllers/AssetItemController.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Controllers;
+
+[ApiController]
+[Route("api/asset-items")]
+public class AssetItemController : ControllerBase
+{
+    private readonly IAssetItemService _assetItemService;
+
+    public AssetItemController(IAssetItemService assetItemService)
+    {
+        _assetItemService = assetItemService;
+    }
+
+    /// <summary>
+    /// Retrieves all AssetItems from the service and returns them in an appropriate HTTP response.
+    /// </summary>
+    /// <returns> a 200 OK response if the operation is successfully and a 400 BadRequest response if any error occurs </returns>
+    [HttpGet]
+    public async Task<IActionResult> GetAllAsync()
+    {
+        return (await _assetItemService.GetAllAsync())
+            .ToErrorOr()
+            .Match<IActionResult>(value => Ok(value), err => BadRequest(err));
+    }
+
+    /// <summary>
+    /// Retrieves an AssetItems with the specific id from the service and returns it in an appropriate HTTP response.
+    /// </summary>
+    /// <param name="id">the unique id of the AssetItem</param>
+    /// <returns> a 200 OK response if the operation is successfully and a 400 BadRequest response if any error occurs </returns>
+
+    [HttpGet($"{{{nameof(id)}:guid}}")]
+    public async Task<IActionResult> GetOneAsync(Guid id)
+    {
+        return (await _assetItemService.GetOneAsync(id)).Match<IActionResult>(
+            value => Ok(JsonSerializer.Serialize(value)),
+            err => BadRequest(err)
+        );
+    }
+
+    /// <summary>
+    /// Updates an existing AssetItem by its ID using the provided update model.
+    /// If the update is successful, returns a 200 OK response. If an error occurs during the update, an appropriate error response is returned.
+    /// </summary>
+    /// <param name="id">the unique id of the AssetItem</param>
+    /// <param name="updateModel">the model containing the updated values for the AssetItem </param>
+
+    [HttpPut($"{{{nameof(id)}:guid}}/{{{nameof(updateModel)}:AssetItem}}")]
```

Comment:
🔧 The `AssetItem` should be transferred in the body and not be necessary in the route
```suggestion
    [HttpPut($"{{{nameof(id)}:guid}}")]
```